xlb/operator/stepper/nse_stepper.py

```diff
@@ -40,6 +40,9 @@ def __init__(self, omega, boundary_conditions=[], collision_type="BGK", forcing_
 
         operators = [self.macroscopic, self.equilibrium, self.collision, self.stream]
 
+        self.boundary_conditions = boundary_conditions
+        self.active_bcs = set(type(bc).__name__ for bc in boundary_conditions)
```

Comment:
Right now the results for sphere example with Regularized inelt and outlet is wrong.  For the current code to work correctly we should replace this line with the following (better to put it  in `_construct_warp` as this is only used for the Warp backened):

`self.active_bcs = set(id_to_bc[bc.id] for bc in boundary_conditions)`

Reply:
we should perhaps also remove `self` and just use `active_bcs` once moved to `_construct_warp`